solidity/test/rewards/TestECDSABackportRewards.js

```diff
@@ -0,0 +1,253 @@
+const {accounts, contract, web3} = require("@openzeppelin/test-environment")
+const {createSnapshot, restoreSnapshot} = require("../helpers/snapshot")
+
+const {expectRevert} = require("@openzeppelin/test-helpers")
+
+const KeepToken = contract.fromArtifact("KeepToken")
+const StackLib = contract.fromArtifact("StackLib")
+const KeepRegistry = contract.fromArtifact("KeepRegistry")
+const BondedECDSAKeepFactoryStub = contract.fromArtifact(
+  "BondedECDSAKeepFactoryStub"
+)
+const KeepBonding = contract.fromArtifact("KeepBonding")
+const MinimumStakeSchedule = contract.fromArtifact("MinimumStakeSchedule")
+const GrantStaking = contract.fromArtifact("GrantStaking")
+const Locks = contract.fromArtifact("Locks")
+const TopUps = contract.fromArtifact("TopUps")
+const TokenStakingEscrow = contract.fromArtifact("TokenStakingEscrow")
+const TokenStaking = contract.fromArtifact("TokenStakingStub")
+const TokenGrant = contract.fromArtifact("TokenGrant")
+const BondedSortitionPoolFactory = contract.fromArtifact(
+  "BondedSortitionPoolFactory"
+)
+const RandomBeaconStub = contract.fromArtifact("RandomBeaconStub")
+const BondedECDSAKeep = contract.fromArtifact("BondedECDSAKeep")
+const ECDSABackportRewards = contract.fromArtifact("ECDSABackportRewards")
+
+const KeepTokenGrant = contract.fromArtifact("TokenGrant")
+
+const BN = web3.utils.BN
+
+const chai = require("chai")
+chai.use(require("bn-chai")(BN))
+const expect = chai.expect
+
+describe("ECDSABackportRewards", () => {
+  let registry
+  let rewardsContract
+  let keepToken
+
+  let tokenStaking
+  let tokenGrant
+  let keepFactory
+  let bondedSortitionPoolFactory
+  let keepBonding
+  let randomBeacon
+  let keepMembers
+
+  const owner = accounts[0]
+
+  const keepSize = 3
+  const numberOfCreatedKeeps = 41
+  const keepCreationTimestamp = 1589408353
```

Comment:
Perhaps `firstKeepCreationTimestamp` given that we are creating them now each 2 hours after another?

Reply:
I've added a comment above. `firstKeepCreationTimestamp += 7200` and then passing in a loop `firstKeepCreationTimestamp` wouldn't be true, because it's a consecutive keep, not the first one.

Reply:
Hmm... I think is is confusing. Maybe we can have:

```  
const keepCreationTimestamp = 1589408352
```

and later:
```
  async function createKeeps() {
    const members = keepMembers.map((m) => m.operator)
    let keepCreationTimestamp = firstKeepCreationTimestamp
    for (let i = 0; i < numberOfCreatedKeeps; i++) {
      await keepFactory.stubOpenKeep(
        owner,
        members,
        tokenStaking.address,
        keepFactory.address,
        keepCreationTimestamp
      )
      keepCreationTimestamp += 7200 // adding 2 hours interval between each opened keep
    }
  }
```
?

Reply:
Ok, done.